src/main/java/com/kapeta/spring/config/KapetaSecurityProviderConfig.java

```diff
@@ -24,13 +35,44 @@ public SecurityFilterChain kapetaFilterChain(HttpSecurity http) throws Exception
                 .securityMatcher(KapetaAuthenticationRestController.PATH_KAPETA_AUTHENTICATION)
                 .authorizeHttpRequests(authorize -> authorize
                         .requestMatchers(KapetaAuthenticationRestController.PATH_KAPETA_AUTHENTICATION).permitAll()
-                        .anyRequest().authenticated()
                 );
+
+        return http.build();
+    }
+
+    @Bean
+    public SecurityFilterChain jwksFilterChain(HttpSecurity http) throws Exception {
+        http
+                .securityMatcher(KapetaJwksRestController.PATH_WELL_KNOWN_JWKS)
+                .authorizeHttpRequests(authorize -> authorize
+                        .requestMatchers(KapetaJwksRestController.PATH_WELL_KNOWN_JWKS).permitAll()
+                );
+
         return http.build();
     }
 
     @Bean
-    public KapetaAuthenticationRestController kapetaAuthenticationRestController() {
-        return new KapetaAuthenticationRestController();
+    public KapetaAuthenticationRestController kapetaAuthenticationRestController(JWKInternalKeyStoreProvider jwkInternalKeyStoreProvider) {
+        return new KapetaAuthenticationRestController(jwkInternalKeyStoreProvider);
+    }
+
+    @Bean
+    public KapetaJwksRestController kapetaJwksRestController(JWKInternalKeyStoreProvider jwkInternalKeyStoreProvider) {
+        return new KapetaJwksRestController(jwkInternalKeyStoreProvider);
+    }
+
+    @Bean
+    @ConditionalOnMissingBean(JWKInternalKeyStoreProvider.class)
+    public JWKInternalKeyStoreProvider jwksPublicKeyProvider() {
+        log.info("Creating JWKS Public Key");
+        try {
+            RSAKey jwk = new RSAKeyGenerator(2048)
+                    .keyUse(KeyUse.SIGNATURE)
+                    .keyID(UUID.randomUUID().toString())
+                    .generate();
+            return () -> new JWKInternalKeyStore("https://auth.kapeta", "https://auth.kapeta", new JWKSet(jwk));
```

Comment:
Let's go with 
```
issuer: 'https://example.auth.kapeta.com',
audience: 'https://example.kapeta.com',
```

Fixing that in nodejs as well

Reply:
https://github.com/kapetacom/codegen-target-nodejs/pull/54

Reply:
Done